engine/Makefile

```diff
@@ -223,11 +225,31 @@ release: $(MODULE)
 
 optimized:CPP_FLAGS += -DNDEBUG
 optimized:OPTS_INTERNAL += -ffast-math -fomit-frame-pointer
-ifneq (${FLAVOR},Darwin)
-  ifneq ($(uname -m),arm64)
+
+# MacOS's default compiler, `clang`, does not support `-march=native` on Apple
+# Silicon chips, and as of 2021-11-22 will return an error like the following
+# at compile time:
+#
+#     clang: error: the clang compiler does not support '-march=native'
+#
+# This conditional preserves the existing default `-march=native` flag on all
+# non-Apple systems, and also on Apple Intel systems, while instead setting
+# `-mcpu=apple-m1` on Apple Silicon systems to enable any M1-specific
+# optimizations that clang supports.
+#
+# In a future release, presumably when there are M2 chips, M3 chips, and so on,
```

Comment:
I fear that even though a new release of clang might support -march=native for M1, you will have again no definition for M2 and it will break again :-)

Reply:
Thoughts and prayers in the mean time. 😂